### ISSUE TITLE
Create support email handler

### ISSUE THREAD

**Issue by hoyes** [camdram/camdram#263]
Write an e-mail handler to process incoming support emails. Should be basically a rewrite of mail-helper.php on Antigone

**Comment by philosophicles** [comment]
The antigone mail-helper.php (and/or the wizardry that precedes it) always had problems with text encoding.  Hence we occasionally get support emails landing in the Issues webpage that we can't read because they're still encoded in e.g. base64.  

Is improving that in the scope of this issue? I had played around with the antigone setup to try to achieve this but never made it work.  I got to the point of thinking that I must be reinventing the wheel somehow - surely somebody had solved the problem elsewhere with free code - but couldn't easily find anything.

**Comment by hoyes** [comment]
The  commit uses Zend's mail library to parse the email. It doesn't quite hand the unencoded base64 text on a plate, but it does parse all the parts and headers automatically, meaning it's easy to detect the base64 header and the encoded content, which is then passed to PHP's base64_decode function. There's a test for doing this at the very bottom of the commit.

I've also tried to improve how emails are routed (inspired partly by how Github does things):
 * Added a 'Return-Path' header, which causes all bounces to be sent to support-bounces@camdram.net rather than back to websupport@. The theory being that this will prevent email loops etc. At the moment these go into a black hole, but the idea is that these will just be logged and won't generate any further emails. There's potential to be cleverer about this eventually (automatically removing old accounts etc).
 * Added the ability for admins to reply to support issues by email. All support replies are forwarded on to all admins and the sender initially. If an issue is 'assigned' to someone, then it'll just be forwarded to them and the sender.
 * You can also 'cc' a support issue, which causes it to be added to the issue page but won't forward it on to anyone.

**Comment by philosophicles** [comment]
Aha, I only skimmed the code and didn't spot the base64 bit - apologies.

Those other improvements sound fantastic :)

**Comment by ajf58** [comment]
Has this been deployed? What's left on this issue?

**Comment by hoyes** [comment]
This is awaiting MX record transferal from Antigone to Tiberius before deployment.

**Comment by CHTJonas** [comment]
We also need to copy/update postfix's virtual aliases file on Tiberius and sync by running the `postmap` command.

**Comment by hoyes** [comment]
So...3 years on and after 30 mins of testing my attempt at an e-mail handler, I'm concluding that:
 * E-mail parsing is hard
 * We are pulling in half of the Zend framework just to do this
 * Other people have already solved this problem.
 * Even if we got this working there would still be things it doesn't do that would be nice to have, such as handling attachments.

Currently looking at https://www.opensupports.com/ ... Looks fairly lightweight and has an extensive API so could even integrate into the existing UI...

**Comment by CHTJonas** [comment]
Looks good. Would this be implemented as a new subdomain eg. support.camdram.net?

**Comment by ajf58** [comment]
What changed? We have a thing that works, and then you said you're waiting on a thing, and now we need a completely different system? I don't understand why we can't use what we have, can you clarify?

**Comment by GKFX** [comment]
The current support system works fine, but it does have the minimum possible set of features. If opensupport works (and continues to work fine with emailed reports) it makes sense just to move.

There’s an existing ticket about not being able to see closed tickets easily; this would probably solve that too.

**Comment by hoyes** [comment]
So for posterity, I went with Brimir in the end because:
 * OsTicket looks a bit dated and doesn't follow modern PHP practices
 * Zammad almost won (and has some nice features for turning Twitter DMs into support tickets etc), but I couldn't get it to work with a postfix pipe / sendmail. The docs seem to imply you have to have a remote imap/pop3 mailbox for it to work properly. It also would've required either installing PostgreSQL or using an out-of-repo MySQL.

We have forked it at https://github.com/camdram/brimir and (with the help of @CHTJonas ) added Camdram Omniauth support. It's set up at https://support.camdram.net/ and I'll send a further e-mail to webteam shortly.
